### PR TITLE
RGB LED Control V2.0

### PR DESCRIPTION
**Pull request by hazemashrafali**
Develop the GPIO Driver and use it to control RGB LED on the Tiva-C board using a push button
**Quick links:**
- [document record](https://drive.google.com/file/d/1Gqlp-tfx9X_fW0qUDJmjotgCF7B2Rq4b/view?usp=sharing)
- [hardware test](https://drive.google.com/file/d/1_yMUOUHz2LZGOUm5uv69fczUm7nUltAR/view?usp=sharing)
- [design document](https://drive.google.com/file/d/1xxXL8iZC1Aq98aMxW1I-qkVO7GRbmFFS/view?usp=sharing)
- [test sheet](https://drive.google.com/file/d/1XiDNCZo9m8Qphhypl5jjcgvLf7kWZskr/view?usp=sharing)

### CODE REVIEW

File: RGB LED Control V2.0/code/Library/std_types.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#define STD_TYPES_H_
+
+typedef enum {FALSE,TRUE}	boolean;
+
+#define LOGIC_HIGH        (1u)
+#define LOGIC_LOW         (0u)
+
+#define NULL_PTR    ((void*)0)
+	
+typedef enum{
```
Comment (Hala-Abdelkader) [inline RGB LED Control V2.0/code/Library/std_types.h:19]: Each driver should have its own enum error state

**Review by Hala-Abdelkader — CHANGES_REQUESTED** [review]

File: RGB LED Control V2.0/code/Library/std_types.h (hunk 1 above)
Comment (hazemashrafali) [reply RGB LED Control V2.0/code/Library/std_types.h:19]: Done, you can check it now

Comment (Hala-Abdelkader) [reply RGB LED Control V2.0/code/Library/std_types.h:19]: Great, well done.👌🏻

Comment (hazemashrafali) [reply RGB LED Control V2.0/code/Library/std_types.h:19]: thank you

**Pull request merged**